.github/workflows/tutorials.yml

```diff
@@ -0,0 +1,97 @@
+name: Jupyter Notebooks
+
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the master branch
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  test:
+    name: Test Jupyter notebooks on ${{ matrix.python-version }} and ${{ matrix.os }} with ${{ matrix.compiler }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      # Prevent all build to stop if a single one fails
+      fail-fast: false
+      matrix:
+        name: [
+          ubuntu-18.04-gcc-9,
+          macOS-10.14-xcode-11.0
+          ]
+
+        include:
+
+          - name: ubuntu-18.04-gcc-9
+            os: ubuntu-18.04
+            compiler: gcc
+            version: "9"
+
+          - name: macOS-10.14-xcode-11.0
+            os: macOS-10.14
+            compiler: xcode
+            version: "11"
+
+        python-version: ['3.6', '3.7']
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install (Linux)
```

Comment:
The install (linux/macos) looks very verbose/complicated, I would assume both linux and OS X come with gcc (ubuntu comes with gcc-7) and clang(OS X always comes with clang), so these shouldn't be necessary. Then can skip mpi and just add it to the CI-mpi and the whole thing is a lot cleaner